snap/snapenv/snapenv.go

```diff
@@ -136,6 +142,26 @@ func basicEnv(info *snap.Info) osutil.Environment {
 		logger.Noticef("cannot determine existence of save data directory for snap %q: %v",
 			info.InstanceName(), err)
 	}
+
+	return env
+}
+
+// appEnv returns the app-level environment variables for a snap.
+func appEnv(info *snap.Info, app *snap.AppInfo) osutil.Environment {
+	env := osutil.Environment{
+		"SNAP_APP_NAME": app.Name,
+	}
+
+	if app.CommonID != "" {
+		env["SNAP_COMMON_ID"] = app.CommonID
+	}
+	if df := app.DesktopFile(); df != "" && osutil.FileExists(df) {
+		env["SNAP_DESKTOP_FILE"] = df
```

Comment:
ok, so this should be then
```suggestion
		env["SNAP_APP_DESKTOP_FILE"] = df
```